adafruit_mcp2515/__init__.py

```diff
@@ -222,24 +222,24 @@ def __init__(
         auto_restart: bool = False,
         debug: bool = False
     ):
-        """[summary]
+        """A common shared-bus protocol.
+        :param ~busio.SPI spi: The SPI bus used to communicate with the MCP2515
+        :param ~digitalio.DigitalInOut cs_pin:  SPI bus enable pin
+        :param int baudrate: The bit rate of the bus in Hz, using a 16Mhz crystal. All devices on the bus must agree on this value. Defaults to 250000.
+        :param bool loopback: Receive only packets sent from this device, and send only to this device. Requires that `silent` is also set to `False`, but only prevents transmission to other devices. Otherwise the send/receive behavior is normal.
+        :param bool silent: When `True` the controller does not transmit and all messages are received, ignoring errors and filters. This mode can be used to “sniff” a CAN bus without interfering. Defaults to `False`.
+        :param bool auto_restart: **Not supported by hardware. An `AttributeError` will be raised if `auto_restart` is set to `True`**
 
-        Args:
-            spi_bus (busio.SPI): The SPI bus used to communicate with the MCP2515
-            cs_pin (digitalio.DigitalInOut): SPI bus enable pin
-            baudrate (int, optional):  The bit rate of the bus in Hz. All devices on the bus must \
-                agree on this value. Defaults to 250000.
-            loopback (bool, optional): When True the rx pin’s value is ignored, and the device \
-                receives the packets it sends. Defaults to False.
-            silent (bool, optional): When True the tx pin is always driven to the high logic level.\
-                 This mode can be used to “sniff” a CAN bus without interfering.. Defaults to False.
-            auto_restart (bool, optional): If True, will restart communications after entering \
-                bus-off state. Defaults to False.
-            debug (bool, optional): If True, will enable printing debug information. Defaults to \
-                False.
+            If `True`, will restart\ communications after entering bus-off state. Defaults to `False`.
+
+        :param bool debug: If `True`, will enable printing debug information. Defaults to `False`.
         """
-        if loopback and silent:
-            raise AttributeError("only loopback or silent mode can bet set, not both")
+
+        if loopback and not silent:
+            raise AttributeError("Loopback mode requires silent to be set")
+        if auto_restart:
+            raise AttributeError("`auto-restart` is not supported by hardware")
```

Comment:
would it be useful to emulate this by restarting at any in_waiting(), receive() or send() call?  is the checking expensive (I could believe that)

Reply:
I think it makes sense to defer implementation until its absence is raised as an issue